### DISCUSSION TITLE
deploying-asp-net-core-applications-to-kubernetes-part-4-creating-a-helm-chart-for-an-aspnetcore-app/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#118] [Announcements]
# deploying-asp-net-core-applications-to-kubernetes-part-4-creating-a-helm-chart-for-an-aspnetcore-app/

In this post I show how you can create a Helm chart for an ASP.NET Core application that you can use to deploy your app to Kubernetes.

https://andrewlock.net/deploying-asp-net-core-applications-to-kubernetes-part-4-creating-a-helm-chart-for-an-aspnetcore-app/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at November 12 2020, 07:05, in reply to [Victor Gladkikh](https://disqus.com/by/AlfeG/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/118#discussioncomment-6059537)</em>

---

<p>I'm going to write a post on this soon I hope, but the short answer is by using an external provider, like Blob Storage/S3, Key Vault, or Redis: <a href="https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/configuration/overview?view=aspnetcore-5.0#persisting-keys-with-redis" rel="nofollow noopener" title="https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/configuration/overview?view=aspnetcore-5.0#persisting-keys-with-redis">https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/configuration/overview?view=aspnetcore-5.0#persisting-keys-with-redis</a></p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at February 22 2021, 10:38, in reply to [jw](https://disqus.com/by/disqus_3VAhYXfFVy/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/118#discussioncomment-6059545)</em>

---

<p>oops, you're right, fixed it - you need to indent the label in your example</p>

[a comment by an automated account is not shown]

**Comment by akhanalcs** [comment]
Hi Andrew,
Thank you for this excellent series, but this post has been very hard for me because looks like I had to install Ingress controller before I could access the app from the browser. And since there was no mention of that in this post, I really struggled with this and still unable to make it work.😩

Can someone please help?

---

I installed Ingress controller following docs [here](https://kubernetes.github.io/ingress-nginx/deploy/#quick-start) for my Docker Desktop's Kubernetes cluster.

I got the cluster info to find the address to add to my hosts file:

<img width="800" alt="image" src="https://github.com/andrewlock/blog-comments/assets/30603497/94c77514-7db9-4864-b326-696a2c381402">

Found the address for  `kubernetes.docker.internal` and added that for `chart-example.local` as well in my hosts file:

<img width="500" alt="image" src="https://github.com/andrewlock/blog-comments/assets/30603497/631a3342-1321-46dd-8066-c46c07301b77">

I then ran deployment:
````
helm upgrade --install test-app-release . --namespace=local --set test-app-api.image.tag="0.1.0" --set test-app-service.image.tag="0.1.0"
````

The Ingress looks like this:

<img width="750" alt="image" src="https://github.com/andrewlock/blog-comments/assets/30603497/6b369a07-117d-4010-9dd0-4cb063681f74">

Finally, I tried to access the web api from my browser, but it doesn't work:

<img width="700" alt="image" src="https://github.com/andrewlock/blog-comments/assets/30603497/e0ad2645-f122-41d8-950b-07717dd13383">

My `values.yaml` file looks like [this](https://github.com/affableashish/k8s-hands-on/blob/main/charts/test-app/values.yaml).

I have no idea of what I'm doing wrong.

**Reply by akhanalcs** [reply]
This was the solution: https://stackoverflow.com/a/77512887/8644294
